Sources/InstantSearchCore/Pagination/PageMap.swift

```diff
@@ -82,8 +82,9 @@ struct PageMap<Item> {
 // MARK: SequenceType
 
 extension PageMap: Sequence {
-  public func makeIterator() -> IndexingIterator<PageMap> {
-    return IndexingIterator(_elements: self)
+  public func makeIterator() -> IndexingIterator<[Item?]> {
+    let elements = (self.startIndex..<self.endIndex).map { self[$0] }
+    return IndexingIterator(_elements: elements)
```

Comment:
I believe this is making a copy of the list elements, which would work. In case of performance issues come up, it might be a good idea to use another approach.

Reply:
Indeed, I see. I believe we can commit to it and monitor for customer feedback as you mentioned. Thanks for your review 🙇 !

Reply:
@aallam I have created this PR to prevent page items copying with each iterator creation, improving performance.